src/private.rs

```diff
@@ -17,55 +17,58 @@ pub use static_assertions as sa;
 ///
 /// - Structs
 ///
-///   [`unique_event_type_and_ver_for_struct`] macro generates function, which
+///   [`unique_event_name_and_ver_for_struct`] macro generates function, which
 ///   returns array with only first occupied entry. The rest of them are
 ///   [`None`].
 ///
 /// - Enums
 ///
-///   [`unique_event_type_and_ver_for_enum`] macro generates function, which
+///   [`unique_event_name_and_ver_for_enum`] macro generates function, which
 ///   glues subtypes arrays into single continues array. First `n` entries are
 ///   occupied, while the rest of them are [`None`], where `n` is the number of
 ///   [`VersionedEvent`]s. As structs deriving [`VersionedEvent`] and enums
-///   deriving [`Event`] have the same output by `__arcana_event_types()` const
+///   deriving [`Event`] have the same output by `__arcana_events()` const
 ///   function, top-level enum variants can have different levels of nesting.
 ///
-///   [`unique_event_type_and_ver_check`] macro generates [`const_assert`]
-///   check, which fails in case of duplicated `event_type` and `ver`.
+///   [`unique_event_name_and_ver_check`] macro generates [`const_assert`]
+///   check, which fails in case of duplicated [`Event::name()`] and
+///   [`Event::ver()`].
 ///
 ///
 /// [`const_assert`]: static_assertions::const_assert
 /// [`Event`]: trait@crate::Event
+/// [`Event::name()`]: trait@crate::Event::name()
+/// [`Event::ver()`]: trait@crate::Event::ver()
 /// [`VersionedEvent`]: trait@crate::VersionedEvent
-pub mod unique_event_type_and_ver {
+pub mod unique_event_name_and_ver {
     #[doc(hidden)]
     #[macro_export]
-    macro_rules! unique_event_type_and_ver_for_struct {
-        ($max_events:literal, $event_type:literal, $event_ver:literal) => {
+    macro_rules! unique_event_name_and_ver_for_struct {
```

Comment:
This macros better move to proc-macro machinery. There is no point to generate a half of an implementation via declarative and another part via proc macros. Better to keep things in a single place asap.